velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -234,46 +234,50 @@ core::PlanNodePtr PlanBuilder::TableWriterBuilder::build(core::PlanNodeId id) {
 
-  // Create column handles.
-  std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
-      columnHandles;
-  for (auto i = 0; i < outputType->size(); ++i) {
-    const auto column = outputType->nameOf(i);
-    const bool isPartitionKey =
-        std::find(partitionBy_.begin(), partitionBy_.end(), column) !=
-        partitionBy_.end();
-    columnHandles.push_back(std::make_shared<connector::hive::HiveColumnHandle>(
-        column,
-        isPartitionKey
-            ? connector::hive::HiveColumnHandle::ColumnType::kPartitionKey
-            : connector::hive::HiveColumnHandle::ColumnType::kRegular,
-        outputType->childAt(i),
-        outputType->childAt(i)));
-  }
+  // If insertHandle_ is not specified, build a HiveInsertTableHandle along with
+  // columnHandles, bucketProperty and locationHandle.
+  if (!insertHandle_) {
+    // Create column handles.
+    std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+        columnHandles;
+    for (auto i = 0; i < outputType->size(); ++i) {
+      const auto column = outputType->nameOf(i);
+      const bool isPartitionKey =
+          std::find(partitionBy_.begin(), partitionBy_.end(), column) !=
+          partitionBy_.end();
+      columnHandles.push_back(std::make_shared<connector::hive::HiveColumnHandle>(
+          column,
+          isPartitionKey
+              ? connector::hive::HiveColumnHandle::ColumnType::kPartitionKey
+              : connector::hive::HiveColumnHandle::ColumnType::kRegular,
+          outputType->childAt(i),
+          outputType->childAt(i)));
+    }
 
-  auto locationHandle = std::make_shared<connector::hive::LocationHandle>(
-      outputDirectoryPath_,
-      outputDirectoryPath_,
-      connector::hive::LocationHandle::TableType::kNew,
-      outputFileName_);
+    auto locationHandle = std::make_shared<connector::hive::LocationHandle>(
+        outputDirectoryPath_,
+        outputDirectoryPath_,
+        connector::hive::LocationHandle::TableType::kNew,
+        outputFileName_);
 
-  std::shared_ptr<HiveBucketProperty> bucketProperty;
-  if (bucketCount_ != 0) {
-    bucketProperty =
-        buildHiveBucketProperty(outputType, bucketCount_, bucketedBy_, sortBy_);
-  }
+    std::shared_ptr<HiveBucketProperty> bucketProperty;
+    if (bucketCount_ != 0) {
+      bucketProperty =
+          buildHiveBucketProperty(outputType, bucketCount_, bucketedBy_, sortBy_);
+    }
 
-  auto hiveHandle = std::make_shared<connector::hive::HiveInsertTableHandle>(
-      columnHandles,
-      locationHandle,
-      fileFormat_,
-      bucketProperty,
-      compressionKind_,
-      serdeParameters_,
-      options_);
+    auto hiveHandle = std::make_shared<connector::hive::HiveInsertTableHandle>(
+        columnHandles,
+        locationHandle,
+        fileFormat_,
+        bucketProperty,
+        compressionKind_,
+        serdeParameters_,
+        options_);
 
-  auto insertHandle =
+    insertHandle_ =
```

Comment:
shouldn't you use this on line 302 now?

Reply:
Yep, forgot to push that change. Adding alongside the `std` qualifier